menagerie/concept-shapes/scripts/mint_trinity.py

```
@@ -0,0 +1,1006 @@
+        print(f"  {abst_name}: {cid[:40]}...")
+
+    # Verify stability of abstractions
+    print("\n  Verifying abstraction stability...")
+    for abst_name, real_keys in abst_real_map.items():
+        memento = copy.deepcopy(abstractions[abst_name])
+        memento["realizations"] = [real_cids[k] for k in real_keys]
+        cid2 = compute_cid(memento)
+        assert cid2 == abst_cids[abst_name], f"UNSTABLE: {abst_name}"
+    print("  All stable.")
+
+    print(f"\n=== Done ===")
```

Ruff (0.15.12)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)